packages/analytics-js/src/components/configManager/ConfigManager.ts

```diff
@@ -1,21 +1,148 @@
+      }
+      state.lifecycle.isStaging.value = isStaging;
+    });
     this.getConfig();
   }
 
-  getConfig() {
-    // Once done call all ready
-    this.onReady();
+  /**
+   * A callback function that is executed once we fetch the source config response.
+   * Use to construct and store information that are dependent on the sourceConfig.
+   * @param res source config response
+   */
+  processConfig(res?: SourceConfigResponse | string) {
+    if (!res || typeof res !== 'object') {
+      throw Error('Unable to fetch source config');
+    }
+    // determine the dataPlane url
+    const dataPlaneUrl = resolveDataPlaneUrl(
+      res.source.dataplanes,
+      state.lifecycle.dataPlaneUrl.value,
+      state.loadOptions.value.residencyServer,
+    );
+    const nativeDestinations: Destination[] =
+      res.source.destinations.length > 0 ? filterEnabledDestination(res.source.destinations) : [];
+
+    // set in the state --> source, destination, lifecycle, reporting
+    batch(() => {
+      // set source related information in state
+      state.source.value = {
+        id: res.source.id,
+      };
+      // set device mode destination related information in state
+      state.destinations.value = nativeDestinations;
+
+      // set application lifecycle state
+      state.lifecycle.activeDataplaneUrl.value = dataPlaneUrl;
+      state.lifecycle.status.value = 'configured';
+
+      // set the values in state for reporting slice
+      state.reporting.isErrorReportingEnabled.value =
+        res.source.config.statsCollection.errorReports.enabled || false;
+      state.reporting.isMetricsReportingEnabled.value =
+        res.source.config.statsCollection.metrics.enabled || false;
+    });
   }
 
-  onReady() {
-    state.lifecycle.status.value = 'configured';
+  /**
+   * A function to fetch source config either from /sourceConfig endpoint
+   * or from getSourceConfig load option
+   * @returns
+   */
+  getConfig() {
+    const sourceConfigOption = state.loadOptions.value.getSourceConfig;
+    if (sourceConfigOption) {
+      // fetch source config from the function
+      const res = sourceConfigOption();
+
+      if (res instanceof Promise) {
+        res
+          .then(pRes => this.processConfig(pRes as SourceConfigResponse))
+          .catch(e => {
+            if (this.hasErrorHandler) {
+              this.errorHandler?.onError(e, 'sourceConfig');
+            }
+          });
+      } else {
+        this.processConfig(res as SourceConfigResponse);
+      }
+      return;
+    }
+
+    // fetch source config from config url API
+    this.httpClient.getAsyncData({
+      url: state.lifecycle.sourceConfigUrl.value,
+      options: {
+        headers: {
+          'Content-Type': undefined,
```

Comment:
Use `Accept` header instead with `application/json` value.

Reply:
In that case, this can be removed from here.

Reply:
we need to unset the default Content-Type header value for thisAPI endpoint

Reply:
Done